### PR TITLE
Feature/migration to sfdx format

### LINKED ISSUE #81: Migration to SFDX Source Format
Related to apex-enterprise-patterns/fflib-apex-common#259

### PR DESCRIPTION
**Pull request by ImJohnMDaniel**
This fixes #81 Migration to SFDX Source Format

The migration script used was the following:
```
mkdir sfdx-source
mkdir sfdx-source/apex-mocks
mkdir sfdx-source/apex-mocks/main
mkdir sfdx-source/apex-mocks/main/classes
mkdir sfdx-source/apex-mocks/test
mkdir sfdx-source/apex-mocks/test/classes
mkdir sfdx-source/apex-mocks/test/classes/mocks

for file in $(ls src/classes/*Test.cls); do git mv $file sfdx-source/apex-mocks/test/classes/; done;
for file in $(ls src/classes/*Test.cls-meta.xml); do git mv $file sfdx-source/apex-mocks/test/classes/; done;
for file in $(ls src/classes/fflib_Mocks.cls); do git mv $file sfdx-source/apex-mocks/test/classes/mocks/; done;
for file in $(ls src/classes/fflib_Mocks.cls-meta.xml); do git mv $file sfdx-source/apex-mocks/test/classes/mocks/; done;
for file in $(ls src/classes/fflib_MyList.cls); do git mv $file sfdx-source/apex-mocks/test/classes/; done;
for file in $(ls src/classes/fflib_MyList.cls-meta.xml); do git mv $file sfdx-source/apex-mocks/test/classes/; done;
for file in $(ls src/classes/*.cls); do git mv $file sfdx-source/apex-mocks/main/classes/; done;
for file in $(ls src/classes/*.cls-meta.xml); do git mv $file sfdx-source/apex-mocks/main/classes/; done;

rm .travis.yml
rm build.xml
rm build.properties.template
rm interfacemocks.properties
rm apex-mocks-generator-*.jar
rm -R src
rm -R lib
```

### CODE REVIEW

File: config/project-scratch-def.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "orgName": "force-di",
```
Comment (daveespo) [inline config/project-scratch-def.json]: s/b orgName: fflib-apex-mocks

File: sfdx-project.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "packageDirectories": [
+        {
+            "path": "sfdx-source/apex-mocks",
```
Comment (daveespo) [inline sfdx-project.json]: I'm not proposing a change here .. just asking a question: How did you arrive at the sfdx-source top level directory name? we usually use 'force-app' for our local application code and we pull in 'fflib-apex-common' and 'fflib-apex-mocks' as top level siblings to the 'force-app' dir

**Review by daveespo — CHANGES_REQUESTED** [review]
@ImJohnMDaniel  -- why did `fflib_Mock.cls` end up in `test/classes/mock` vs. `test/classes`?

File: config/project-scratch-def.json (hunk 1 above)
Comment (ImJohnMDaniel) [reply config/project-scratch-def.json]: fixed

File: sfdx-project.json (hunk 2 above)
Comment (ImJohnMDaniel) [reply sfdx-project.json]: @daveespo -- this is the basic format that comes from the "Falcon Template" approach of DX project setup.  It allows you to add the other source directories related to the project like `sfdx-source/unpackaged` and `sfdx-source/untracked`.  I have found it to be a great setup in the projects that I have worked with.  I am certainly open to other suggestions though.

**Comment by ImJohnMDaniel** [comment]
@daveespo -- regarding your question 
```
why did fflib_Mock.cls end up in test/classes/mock vs. test/classes?
```
I moved it there simply to have some basic separation of concerns.  The fflib_Mocks class is basically a mock class exclusively used by test classes but it is not a test class by itself.

**Review by daveespo — APPROVED** [review]
@ImJohnMDaniel -- looks good .. +1 for merge

**Review by afawcett — APPROVED** [review]
+1

**Comment by DanielTrontsios** [comment]
@ImJohnMDaniel Why did the apex mock generators get fully deleted here? should they not be used anymore?

**Pull request merged**